scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -101,6 +102,22 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
   def withHotKeyFanout(hotKeyFanout: Int): SCollectionWithHotKeyFanout[K, V] =
     new SCollectionWithHotKeyFanout(this, Right(hotKeyFanout))
 
+  /**
+   * Apply Zstd compression to the _value side only_ of this SCollection, using the provided Zstd
+   * dictionary. Dictionary must have been trained on exactly type `V`.
+   *
+   * @param dict
+   *   The Zstd dictionary trained on type `V`.
+   */
+  def setZstdDictionaryForValue(dict: Array[Byte]): SCollection[(K, V)] = {
```

Comment:
Going to reconsider this API. Using it in conjunction with setCoder is going to be confusing since they both mutate the underlying PCollection